### PR TITLE
feat: Use ordinal measures with maps

### LINKED ISSUE #734: Colored Map with Ordinal Scale Measure
### User Story
As a user I would like to show a colored map from my ordinal scale measure (e.g. low, medium, high).\

### Definition of Done
Ordinal measures can be used as color mapping for map chart layers – with `schema:name` and `schema:description` appearing in the tooltip and colors that are pre-defined based on potential `schema:color`.

### Additional Info
* [Example dataset](https://test.visualize.admin.ch/en/browse/dataset/https%3A%2F%2Fenvironment.ld.admin.ch%2Ffoen%2Ftest_ordinal_507%2F7)
* Take the following into account:
  * Sort order (i.e. severity) according to `schema:position`
  * Typical attributes shall be displayed on hovering (currently, NaNs see [example](https://test.visualize.admin.ch/en/v/AP2HkMGFzw8w))
    * `schema:name @lang`
    * `schema:description @lang`
  * For the corresponding table view `schema:name` will be enough (same as implemented already for nominal key dimensions in table view)
* Color of levels can differ case-by-case, i.e. danger case here [green, yellow and red](https://www.naturgefahren.ch) while non danger may select other colors

image
image

### LINKED ISSUE #781: "automatic" color modification when editing a visualization
**Describe the bug**
When editing an existing visualization (button "Copier et éditer cette visualisation"), and then when clicking on the "Couleur" parameter, once the filters are uploaded the colors change

**To Reproduce**
1. Open one of those 2 visualizations:
https://test.visualize.admin.ch/fr/v/3cbX1vtyQtQQ?dataSource=Test
https://test.visualize.admin.ch/fr/v/CutNfGjCNnpq?dataSource=Test
2. Click on "Copier et éditer cette visualisation"
3. Click on the left on "Couleur"
4. The colors change once the filters are uploaded

**Environment (please complete the following information):**
 - Visualize TEST v3.9.6
 - Browser Firefox 105.0.3

### PR DESCRIPTION
**Pull request by bprusinowski**
Closes #734.
Fixes #781.

This PR introduces the usage of ordinal measures in maps (both in area and symbol layers). Additionally, we now fetch optional colors assigned to dimension values and use them as default, initial color values in color mappings. It also adds title for non-hierarchical map legend.

### How to test
1. Go to `/en/browse/dataset/https%3A%2F%2Fenvironment.ld.admin.ch%2Ffoen%2Ftest_ordinal_507%2F7?dataSource=Int`
2. Select map chart type.
3. Select area layer.
4. See that `ord1` (ordinal measure) is selected and it's possible to select other ordinal measures.
5. See that the colors are custom-defined (green-red).
6. Select symbol layer.
7. Enable the layer.
8. See that in this **_extreme_** example we don't have any numerical measure so the radius of the symbols is fixed and shared across all symbols.
9. Change color mapping and see that it's possible to choose ordinal measure to color the symbols.

### CODE REVIEW

**Comment by ptbrowne** [comment]
I think it'd be good to have an E2E test for this feature :)

File: app/rdf/query-literals.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  ids: NamedNode[];
+  sparqlClient: ParsingClient;
+  predicate: NamedNode<string>;
+}) {
+  return batchLoad<ResourceLiteral, NamedNode>({
+    ids,
+    sparqlClient,
+    buildQuery: (values) => buildResourceLiteralQuery({ values, predicate }),
+  });
+}
```
Comment (ptbrowne) [inline app/rdf/query-literals.ts:45]: Cool 👍

File: app/configurator/components/chart-options-selector.tsx [hunk 2]
```diff
@@ -688,6 +691,10 @@ const ChartFieldColorComponent = ({
     "color",
     "componentIri",
   ]) as string | undefined;
+  const colorComponent = [
+    ...dataSetMetadata.dimensions,
+    ...dataSetMetadata.measures,
+  ].find((d) => d.iri === colorComponentIri);
```
Comment (ptbrowne) [inline app/configurator/components/chart-options-selector.tsx:694]: At some point, we'll want to make a useDimensionByIri hook backed by a provider, we do too much of this pattern everywhere 😅

File: app/charts/map/map-state.tsx [hunk 3]
```diff
@@ -495,10 +495,15 @@ const useMapState = (
   });
 
   const radiusScale = useMemo(() => {
-    return scaleSqrt()
-      .domain([0, symbolLayerState.dataDomain[1]])
-      .range([0, 24]);
-  }, [symbolLayerState.dataDomain]);
+    // Measure dimension is undefined.
+    if (symbolLayerState.dataDomain[1] === undefined) {
+      return scaleSqrt().domain([0, 0]).range([0, 24]).unknown(12);
```
[the remark this answers is not in the document]
Comment (bprusinowski) [reply app/charts/map/map-state.tsx:471]: Yes, I think it makes sense when the user wants to just show the color, while the radius is irrelevant, as it the case in example shared [here](https://gitlab.ldbar.ch/bafu/umweltdatenkiosk-planning/-/issues/374) (last image)

[the remark this answers is not in the document]
Comment (bprusinowski) [reply app/charts/map/map-state.tsx:471]: I think in this case range could also be [0, 0], or maybe even `scaleSqrt().unknown(12)` would work in this case, I will check

File: app/charts/map/map-state.tsx [hunk 4]
```diff
@@ -495,10 +495,15 @@ const useMapState = (
   });
 
   const radiusScale = useMemo(() => {
-    return scaleSqrt()
-      .domain([0, symbolLayerState.dataDomain[1]])
-      .range([0, 24]);
-  }, [symbolLayerState.dataDomain]);
+    // Measure dimension is undefined.
```
Comment (ptbrowne) [inline app/charts/map/map-state.tsx:469]:
```suggestion
    // Measure dimension is undefined. Can be useful when the user want to encode only
    // the color of symbols, and the size is irrelevant.
```

File: app/charts/map/map-state.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ptbrowne) [reply app/charts/map/map-state.tsx:471]: OK, thanks I added a suggestion for the comment https://github.com/visualize-admin/visualization-tool/pull/778/files/1249a3033558f6c51ad7faee1016e284bb97cb7a..4200197986c19e37998d8e9b89aaa468bbbddd93#r995594514

[the remark this answers is not in the document]
Comment (bprusinowski) [reply app/charts/map/map-state.tsx]: Done

**Review by ptbrowne — APPROVED** [review]

**Pull request merged**